packages/components/inputs/async-select-input/src/async-select-input.tsx

```diff
@@ -370,7 +377,13 @@ const AsyncSelectInput = (props: TAsyncSelectInputProps) => {
               ...props.components,
             } as ReactSelectAsyncProps['components']
           }
-          menuIsOpen={props.isReadOnly ? false : undefined}
+          menuIsOpen={
+            props?.menuIsOpen && !props.isReadOnly
+              ? props.menuIsOpen
+              : props.isReadOnly
+              ? false
+              : undefined
+          }
```

Comment:
I would suggest a simpler condition:

```suggestion
          menuIsOpen={
            props.isReadOnly ? false : props.menuIsOpen
          }
```